src/System.Linq/tests/OrderByTests.cs

```diff
@@ -441,5 +443,93 @@ public void TakeOne(IEnumerable<int> source)
             }
             Assert.Equal(1, count);
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
Rather than making this specific to Windows, can you just use Array.Sort to create the expected results?

Also, we have a helper we try to use in our tests to switch the current culture:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/CoreFx.Private.TestUtilities/System/ThreadCultureChange.cs
e.g.
https://github.com/dotnet/corefx/blob/3b6ec2ba944fb72f921f95b99b6ebf2e7bb3dc5a/src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrentInfo.cs#L26-L29

Thanks.

Reply:
Hmmm. Kind of in two minds about using Array.Sort up front, as if we just get the same ordering for both test arrays then it negates the point of the test. But as it would do the "right thing" (arguably, obviously!) under windows, and make the tests pass under linux then I guess I'll go with it.

Also due to my multiple culture changes I'm not sure if `ThreadCultureChange` is the right choice here? I'd probably prefer just to leave my existing `finally` block as I think it less obscures the point of this test, which is ensuring that it's on the first `MoveNext` that the Culture is locked in? Thoughts?

Reply:
> Also due to my multiple culture changes I'm not sure if ThreadCultureChange is the right choice here?

We use it as such in other places; I think it'd be good to do so here, too.  If nothing else, it helps to visualize highlight via scoping to which operation each culture is being applied.